public/src/main/java/com/revenuecat/purchases/models/PurchaseOption.kt

```diff
@@ -6,12 +6,12 @@ import android.os.Parcelable
  * A purchase-able entity for a subscription product.
  */
 interface PurchaseOption : Parcelable {
-    /**
-     * If this PurchaseOption represents a Google subscription base plan, this will be the base plan ID.
-     * If it represents an offer, it will be the offer ID.
-     * Not applicable for Amazon subscriptions.
-     */
-    val id: String
+
+    // TODO BC5 javadoc
+    val basePlanId: String
+
+    // TODO BC5 javadoc
+    val offerId: String?
```

Comment:
I think this is breaking the abstraction for PurchaseOption, or at least complicating it more. In the end it's tying it again to baseplans/offers.
In this case I'd suggest to just make an idea composed by both (so just `{baseplanId}` or `{baseplanId}:{offerId}`

Reply:
Agreed